nengo/tests/test_connection.py

```diff
@@ -613,6 +613,45 @@ def test_slicing_function(Simulator, plt, seed):
     assert np.allclose(w, y, atol=0.1)
 
 
+def test_advanced_indexing(Simulator, plt, seed):
+    N = 100
+
+    model = nengo.Network()
+    with model:
+        u = nengo.Node([-1, 1])
+        a = nengo.Ensemble(n_neurons=N, radius=3, dimensions=1)
+        b = nengo.Ensemble(n_neurons=N, radius=3, dimensions=1)
+        c = nengo.Ensemble(n_neurons=N, radius=3, dimensions=2)
+        d = nengo.Ensemble(n_neurons=N, radius=3, dimensions=2)
+        nengo.Connection(u[[0, 1]], a[[0, 0]])
+        nengo.Connection(u[[1, 1]], b[[0, 0]])
+        nengo.Connection(u[[0, 1]], c[[0, 1]])
+        nengo.Connection(u[[1, 1]], d[[0, 1]])
+
+        a_probe = nengo.Probe(a, synapse=0.01)
+        b_probe = nengo.Probe(b, synapse=0.01)
+        c_probe = nengo.Probe(c, synapse=0.01)
+        d_probe = nengo.Probe(d, synapse=0.01)
+
+    with nengo.Simulator(model) as sim:
+        sim.run(0.4)
+
+    t = sim.trange()
+    a_data = sim.data[a_probe]
+    b_data = sim.data[b_probe]
+    c_data = sim.data[c_probe]
+    d_data = sim.data[d_probe]
+
+    plt.plot(t, a_data)
+    assert np.allclose(a_data[100:], [0], atol=0.3)
```

Comment:
These tolerances feel pretty big to me; I'll look at the plots when I rereview, but try a larger synapse value on the probes to see if the decoded value is more stable?